jetbrains-core/tst/software/aws/toolkits/jetbrains/services/codewhisperer/codescan/CodeWhispererRubyCodeScanTest.kt

```diff
@@ -0,0 +1,246 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.codewhisperer.codescan
+
+import com.intellij.openapi.vfs.VirtualFile
+import kotlinx.coroutines.runBlocking
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.junit.jupiter.api.assertThrows
+import org.mockito.kotlin.any
+import org.mockito.kotlin.spy
+import org.mockito.kotlin.stub
+import software.aws.toolkits.jetbrains.services.codewhisperer.codescan.sessionconfig.CodeScanSessionConfig
+import software.aws.toolkits.jetbrains.services.codewhisperer.codescan.sessionconfig.RubyCodeScanSessionConfig
+import software.aws.toolkits.jetbrains.utils.rules.PythonCodeInsightTestFixtureRule
+import software.aws.toolkits.telemetry.CodewhispererLanguage
+import java.io.BufferedInputStream
+import java.util.zip.ZipInputStream
+import kotlin.io.path.relativeTo
+import kotlin.test.assertNotNull
+
+class CodeWhispererRubyCodeScanTest : CodeWhispererCodeScanTestBase(PythonCodeInsightTestFixtureRule()) {
+    private lateinit var testRuby: VirtualFile
+    private lateinit var utilsRuby: VirtualFile
+    private lateinit var helperRuby: VirtualFile
+    private lateinit var sessionConfigSpy: RubyCodeScanSessionConfig
+
+    private var totalSize: Long = 0
+    private var totalLines: Long = 0
+
+    @Before
+    override fun setup() {
+        super.setup()
+        setupRubyProject()
+        sessionConfigSpy = spy(CodeScanSessionConfig.create(testRuby, project) as RubyCodeScanSessionConfig)
+        setupResponse(testRuby.toNioPath().relativeTo(sessionConfigSpy.projectRoot.toNioPath()))
+
+        mockClient.stub {
+            onGeneric { createUploadUrl(any()) }.thenReturn(fakeCreateUploadUrlResponse)
+            onGeneric { createCodeScan(any(), any()) }.thenReturn(fakeCreateCodeScanResponse)
+            onGeneric { getCodeScan(any(), any()) }.thenReturn(fakeGetCodeScanResponse)
+            onGeneric { listCodeScanFindings(any(), any()) }.thenReturn(fakeListCodeScanFindingsResponse)
+        }
+    }
+
+    @Test
+    fun `test createPayload`() {
+        val payload = sessionConfigSpy.createPayload()
+        assertNotNull(payload)
+        assertThat(payload.context.totalFiles).isEqualTo(3)
```

Comment:
same here, better to assert the contents of the list

Reply:
Payload does have a parameter like number of files present but does not have exact file names. So I am not able to check the fileNames for all files present but we can do this for scanned files though. Correct me If I am wrong @Will-ShaoHua 
Thank you


![image](https://github.com/aws/aws-toolkit-jetbrains/assets/141967714/aeaa9dfc-c062-4d03-bbb1-f58801f7644b)

Reply:
oh nm, I thought it's a list but it's not

Reply:
thanks for the clarification